### PR TITLE
M/no sort merger

### PR DESCRIPTION
**Pull request by mfelsche**

### CODE REVIEW

File: sql/src/test/java/io/crate/operation/merge/AbstractBucketMergerTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.crate.core.collections.Bucket;
+import io.crate.core.collections.CollectionBucket;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class AbstractBucketMergerTest {
+    protected List<ListenableFuture<Bucket>> createBucketFutures(List<Object[]> ... buckets) {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/operation/merge/AbstractBucketMergerTest.java]: can you make this static and move into a BucketTestingHelpers class instead of subclassing?

File: sql/src/main/java/io/crate/operation/projectors/SortingTopNProjector.java [hunk 2]
```diff
@@ -96,15 +98,17 @@ public void startProjection() {
 
     @Override
     public synchronized boolean setNextRow(Row row) {
-        if (spare == null) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/projectors/SortingTopNProjector.java]: can you revert the changes in the SortingTopNProjector and instead create a separate OrderByPosition for the BucketMerger which is Row based and leave the Object[][] stuff here as is?

File: core/src/main/java/io/crate/core/collections/RowBucket.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.core.collections;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+public class RowBucket implements Bucket {
```
Comment (mfussenegger) [inline core/src/main/java/io/crate/core/collections/RowBucket.java]: delete this once the TopN change is reverted as it won't be needed anymore

File: sql/src/main/java/io/crate/operation/merge/NonSortingBucketMerger.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            }
+        };
+        for (ListenableFuture<Bucket> bucketFuture : buckets) {
+            Futures.addCallback(bucketFuture, callback);
+        }
+        try {
+            countDownLatch.await();
+            Throwable caught = exception.get();
+            if (caught == null) {
+                finish();
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/merge/NonSortingBucketMerger.java]: shouldn't finish be called by the upstream and not inside merge?

File: sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java [hunk 5]
```diff
@@ -653,11 +653,11 @@ public void testInsertFromSubQueryPartitionedTableCustomSchema() throws Exceptio
         ensureGreen();
         refresh();
 
-        execute("select * from custom.source order by zipcode");
+        execute("select * from custom.source order by name");
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java]: why has this been changed?

Comment (mfelsche) [inline sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java]: ordering by zipcode is not deterministic as there are two rows with the same zipcode

**Comment by mfussenegger** [comment]
ack
